sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -866,37 +866,36 @@ static int sof_card_dai_links_create(struct device *dev,
 			      dmic_component, 1, dmic_init, NULL);
 	}
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_COMMON_HDMI_CODEC)
+#if IS_ENABLED(CONFIG_SND_HDA_CODEC_HDMI)
 	/* HDMI */
-	if (hdmi_num > 0) {
-		idisp_components = devm_kcalloc(dev, hdmi_num,
-						sizeof(*idisp_components),
+	for (i = 0; i < hdmi_num; i++) {
+		struct snd_soc_dai_link_component *idisp_components;
+
+		idisp_components = devm_kzalloc(dev, sizeof(*idisp_components),
 						GFP_KERNEL);
 		if (!idisp_components)
 			return -ENOMEM;
-	}
 
-	for (i = 0; i < hdmi_num; i++) {
```

Comment:
Unless there's a real practical reason for this, please, revert. Please keep the array.

Reply:
see above reason, thanks!